aspnet/mvc/controllers/routing.rst

```diff
@@ -1,8 +1,140 @@
-=========================================
+By `Rachel Appel`_
 
-.. include:: /../common/stub-notice.txt
+In this article:
 
-.. _issue: https://github.com/aspnet/Docs/issues/117
+.. contents:: Sections
+  :local:
+  :depth: 1
+
+Relationship to middleware
+--------------------------
+
+In the early days of the web you didn’t think about routing, as most web sites or applications simply referenced folders and files on a disk. If you had a URL for ``http://contoso.com/movies/GoneWithTheWind.htm``, then there was a folder called ``movies`` with a file called ``GoneWithTheWind.htm`` on the disk. With modern web frameworks like MVC you do not have to structure our content or application exactly how the world will see it. You can present it however you choose, and organize your software in the way that makes sense. `Routing <https://docs.asp.net/en/latest/fundamentals/routing.html>`_ is what you use to connect our code to the URL and features our application exposes. 
+
+Routing is part of ASP.NET Core middleware. Middleware is a component that is assembled into an application pipeline to handle requests and responses. It's the middleware that processes the request until it gets to routing, which then hands over the request to MVC. MVC then selects an action method based on data that comes from routing. After that point it's up to the developer to add code to the action method to retrieve and modify data, or call services. 
+
+Startup code and relationship to MVC route handler
+--------------------------------------------------
+
+To configure routing, add the following code to the ``Configure`` method in the ``Startup.cs`` file. The code maps a route named ``default`` to a controller named ``Home`` and action method named ``Index``. You configure this by indicating which controllers, action methods, and parameters you want recognized as a route in the ``template`` argument of the ``MapRoute`` method, as shown below: 
+
+.. code-block:: c# 
+
+  app.UseMvc(routes =>
+  {
+    routes.MapRoute(
+      name: "default",
+      template: "{controller=Home}/{action=Index}/{id?}");
+  });
+  
+The route template defines the route as well as default values, in this example, ``Home`` for the controller, ``Index`` for the action method, and optionally ``id`` as a URL parameter. Routes defined in the ``Startup.cs`` file are conventional routes. 
+
+Conventional routing
+--------------------
+
+Routing first matches the URL to a route template, then selects a controller and action to route the request. You may define routes and route constraints on both controllers and actions. In conventional routing, you define a route that generates the action and controller tokens. The value of those tokens are used to select an action and controller by name. For example, the URL ending in ``/home/index/`` matches the route template for ``{controller}/{action}``. Conventional routing must be configured in the ``Startup`` class, as shown in the previous code sample. 
```

Comment:
> In conventional routing, you define a route that generates the action and controller tokens. The value of those tokens are used to select an action and controller by name.

We should call these route values instead of tokens. There's another concept in routing called data tokens, and these are not that.